src/wwwroot/controllers/TemplateCtrl.js

```diff
@@ -39,38 +43,57 @@
 
     function load(templateId) {
       $scope.loadInProgress = true;
-
-      return templates.getTemplateWithBody(templateId).then(function (result) {
-        $scope.template.fromName = result.from_name;
-        $scope.template.fromEmail = result.from_email;
-        $scope.template.subject = result.subject;
-        $scope.template.name = result.name;
-        $scope.template.id = result.id;
-        $scope.template.content = result.body;
+      return templates.getTemplate(templateId).then(function (template) {
+        $scope.template.fromName = template.from_name;
+        $scope.template.fromEmail = template.from_email;
+        $scope.template.subject = template.subject;
+        $scope.template.name = template.name;
+        $scope.template.id = template.id;
+        $scope.isHtmlTemplate = template.bodyType == "rawHtml" || (template.bodyType == "empty" && !$rootScope.forceMsEditor);
+        if (!$scope.isHtmlTemplate) {
+          $scope.template.content = null;
+        } else {
+          return templates.getTemplateBody(templateId).then(function(templateBody) {
+            $scope.template.content = templateBody.html;
+          });
+        }
       }).finally(function () {
         $scope.loadInProgress = false;
       });
     }
 
     function save() {
       $scope.saveInProgress = true;
-      return templates.save({
+      var templateModel = {
         from_name: $scope.template.fromName,
         from_email: $scope.template.fromEmail,
         subject: $scope.template.subject,
         body: $scope.template.content,
         name: $scope.template.name,
         id: $scope.template.id
-      }).finally(function () {
-        $scope.saveInProgress = false;
```

Comment:
It seems that you have lost this _finally_ block.

Reply:
Yes, sorry thx for the catch